File: Documentation/3.0/Samples/csharp/Raven.Documentation.Samples/FileSystem/ClientApi/Session/OpeningSession.cs
namespace Raven.Documentation.Samples.FileSystem.ClientApi.Session
{
	using System.Threading.Tasks;
	using Client.Document;
	using Client.FileSystem;

	public class OpeningSession
	{
		private IFilesStore store = null;

		interface IInterface
		{
			#region open_session_1
			// Open a session for a 'default' file system configured in 'FilesStore'
			IAsyncFilesSession OpenAsyncSession();

			// Open a session for a specified file system
			IAsyncFilesSession OpenAsyncSession(string filesystem);

			// Open a session for a specified file system with custom API Key / Credentials
			IAsyncFilesSession OpenAsyncSession(OpenFilesSessionOptions sessionOptions);
			#endregion
		}

		public async Task Foo()
		{
			FilesStore store = null;

			#region open_session_2
			store.OpenAsyncSession(new OpenFilesSessionOptions()
			{
				FileSystem = store.DefaultFileSystem
			});
			#endregion

			string filesystem = null;

			#region open_session_3
			store.OpenAsyncSession(new OpenFilesSessionOptions()
			{
				FileSystem = filesystem
			});
			#endregion

			#region open_session_4
			using (IAsyncFilesSession session = store.OpenAsyncSession())
			{
				// code here
			}
			#endregion
		}
	}
}

File: Documentation/3.5/Samples/python/clientapi/session/OpeningSession.py
from pyravendb.store.document_store import documentstore


class OpeningSession(object):
    @staticmethod
    def open_session():
        store = documentstore()
        store.initialize()

        # region open_session_1
        # Open session for a 'default' database configured in 'documentstore'
        with store.open_session() as session:
            session.load("doc/1")
            # code here

        # Open session for a specific database
        with store.open_session(database="Your database") as session:
            session.load("doc/2")
            # code here

        # Open session for a specific database with api_key
        with store.open_session(database="Your database", api_key="API_KEY") as session:
            session.load("doc/3")
            # code here

        # endregion
